openshift/templates/template-mongodb-backup.yaml

```diff
@@ -0,0 +1,222 @@
+                        UNHEALTHY_STR="UNHEALTHY";
+                        UNHEALTHY_STR=${1/HEALTHY/$UNHEALTHY_STR};
+                        mv /var/lib/mongodb-backup/$HEALTHY_STR /var/lib/mongodb-backup/$UNHEALTHY_STR;
+                        ALL_BACKUPS_HEALTHY=false;
+                        curl -X POST -H "Content-Type: application/json" --data "{\"username\":\"BakBot\",\"icon_emoji\":\":robot:\",\"text\":\"@all EPIC Mongo backup FAILURE. Backups are currently paused until the issue is fixed. $MESSAGE Please see documentation https://github.com/bcgov/esm-server/tree/dev/openshift\"}" $ROCKETCHAT_BACKUP_DB_WEBHOOK;
+                      };
+                      alertWarning () {
+                        MESSAGE="$1";
+                        curl -X POST -H "Content-Type: application/json" --data "{\"username\":\"BakBot\",\"icon_emoji\":\":robot:\",\"text\":\"@all EPIC Mongo backup WARNING - $MESSAGE Please see documentation https://github.com/bcgov/esm-server/tree/dev/openshift\"}" $ROCKETCHAT_BACKUP_DB_WEBHOOK;
+                      };
+                      checkDumps () {
+                        DUMP_DIRS=$(ls /var/lib/mongodb-backup);
+                        for DUMP_DIR in $DUMP_DIRS; do
+                          for ADMIN_FILE in "${adminDirFiles[@]}"; do
+                            if [ ! -f /var/lib/mongodb-backup/$DUMP_DIR/admin/$ADMIN_FILE ]; then
+                              alertBadDump "$DUMP_DIR" "A previous backup did not complete successfully. One or more files in the admin directory are missing.";
+                              break;
+                            fi;
+                          done;
+                          for ESM_FILE in "${esmDirFiles[@]}"; do
+                            if [ ! -f /var/lib/mongodb-backup/$DUMP_DIR/esm/$ESM_FILE ]; then
+                              alertBadDump "$DUMP_DIR" "A previous backup did not complete successfully. One or more files in the esm directory  are missing.";
+                              break;
+                            fi;
+                          done;
+                          if [ "$(du -sb /var/lib/mongodb-backup/$DUMP_DIR | cut -f1)" -gt $MONGO_BACKUP_SIZE_UPPER_LIMIT ]; then
+                            alertWarning "Disk usage is approaching capacity. Please transfer backups to larger PVC or decrease the number of backups to keep.";
+                          fi;
+                          if [ "$(du -sb /var/lib/mongodb-backup/$DUMP_DIR | cut -f1)" -lt $MONGO_BACKUP_SIZE_LOWER_LIMIT ]; then
+                            alertBadDump "$DUMP_DIR" "A previous backup did not complete successfully. Backup size was lower than normal.";
+                          fi;
+                        done;
+                      };
+                      checkDumps;
+                      if [ "$ALL_BACKUPS_HEALTHY" = true ]; then
+                        DIR=/var/lib/mongodb-backup/dump--`date +%F--%H-%M-%S--%Z`--HEALTHY;
```

Comment:
How about `date +%Y%m%d%H%M%S%Z` ?  In the documentation file it talks about ':' characters and problems with rsync.  Was this already addressed in this line?  Looks like it.  Let's update the docs if so.